cosmwasm/ibc-union/app/ucs03-zkgm/src/contract.rs

```diff
@@ -1471,6 +1472,54 @@ fn execute_call(
         )
         .map_err(|_| ContractError::UnableToValidateCallTarget)?;
 
+    let predicted_address = predict_call_proxy_account(
+        deps.as_ref(),
+        &env,
+        path,
+        packet.destination_channel_id,
+        call.sender.clone().into(),
+    )?;
+
+    let mut response = Response::<cosmwasm_std::Empty>::new();
+
+    if predicted_address == contract_address
+        && deps
+            .querier
+            .query_wasm_contract_info(&contract_address)
+            // REVIEW: Is this sound? Should there be a more sophisticated check?
+            .is_err()
+    {
+        let Config { dummy_code_id, .. } = CONFIG.load(deps.storage)?;
+
+        response = response
+            .add_message(WasmMsg::Instantiate2 {
+                admin: Some(env.contract.address.to_string()),
+                code_id: dummy_code_id,
+                label: proxy_account_label(
+                    path,
+                    packet.destination_channel_id,
+                    call.sender.clone().into(),
+                ),
+                msg: to_json_binary(&cosmwasm_std::Empty {})?,
+                funds: vec![],
+                salt: proxy_account_salt(
+                    path,
+                    packet.destination_channel_id,
+                    call.sender.clone().into(),
+                )
+                .get()
+                .to_vec()
+                .into(),
+            })
+            .add_event(
+                Event::new("create_proxy_event")
+                    .add_attribute("path", path.to_string())
+                    .add_attribute("channel_id", packet.destination_channel_id.to_string())
+                    .add_attribute("owner", call.sender.to_string())
+                    .add_attribute("address", predicted_address),
+            );
```

Comment:
aren't we missing the migrate msg to switch from dummy_code_id to proxy_code_id?

Reply:
also, don't we want to set the proxy admin of himself such that the owner can dispatch an updateadmin?